storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -570,7 +575,11 @@
                                                          (tasks-fn out-stream-id values))
                                              rooted? (and message-id has-ackers?)
                                              root-id (if rooted? (MessageId/generateId rand))
-                                             out-ids (fast-list-for [t out-tasks] (if rooted? (MessageId/generateId rand)))]
+                                             out-ids (fast-list-for [t out-tasks] (if rooted? (MessageId/generateId rand)))
+                                             sampler? (sampler)]
+                                         (when sampler?
+                                           (.notify rate-tracker (count out-tasks))
+                                           (stats/update-stats-throughput! (:stats executor-data) out-stream-id (.reportRate rate-tracker)))
```

Comment:
This is calculating the rate for the spout ignoring the stream-id, and then storing it into the other stats saying it is for a specific stream id.  That does not work.  Either we have to drop the stream-id from the reported stats or we have to calculate it correctly per stream id.